### ISSUE TITLE
New flake install instructions

### ISSUE THREAD

**Issue by bestlem** [LnL7/nix-darwin#740]
They don"t work.

The documentation needs to also say when editing flake.nix

Edit the two "simple" in the end of the file to the first part of the machine name (not including any .local or other domain) m

9OK mine now also fails due to lack of all the build users - but I think that is a bug #681 The default for new users should not need the extra line that is recommennded. Begginer documentation MUST be exact so the user can copy and paste the example and comments like edit this file and not good but can be dealt with but asking a begginner to set a variable does not work as where in the file does that edit get done.
And the response https://github.com/LnL7/nix-darwin/pull/681#issuecomment-1584586205 looks like it is wrong to me as the steps listed here don"t work without the added line to suppress the check.

**Comment by emilazy** [comment]
> The documentation needs to also say when editing flake.nix
> 
> Edit the two "simple" in the end of the file to the first part of the machine name (not including any .local or other domain) m

The flake installation instructions say the following:

> Make sure to replace all occurrences of `simple` with your short hostname which you can find by running `hostname -s`.

Could you suggest how we could make this more clear for beginners?

> 9OK mine now also fails due to lack of all the build users - but I think that is a bug #681 The default for new users should not need the extra line that is recommennded. Begginer documentation MUST be exact so the user can copy and paste the example and comments like edit this file and not good but can be dealt with but asking a begginner to set a variable does not work as where in the file does that edit get done. And the response [#681 (comment)](https://github.com/LnL7/nix-darwin/pull/681#issuecomment-1584586205) looks like it is wrong to me as the steps listed here don"t work without the added line to suppress the check.

Are you using the Determinate Systems experimental [nix-installer](https://github.com/DeterminateSystems/nix-installer)? That's not an officially-supported way to install Nix, and has some issues currently, although it's good software and we do try to support it. They tried avoiding user creation on macOS as an experiment, but it [doesn't work properly](https://github.com/DeterminateSystems/nix-installer/issues/521) and they [disabled it](https://github.com/DeterminateSystems/nix-installer/pull/524). That was released as [v0.10.0](https://github.com/DeterminateSystems/nix-installer/releases/tag/v0.10.0), so if you do a clean install with the same installer now you shouldn't have the same problem.

We could disable the build users check unconditionally, but it may make things more brittle for people using the official installer or the latest version of Determinate Systems' unofficial one. `auto-allocate-uids` doesn't really work on macOS yet, unfortunately, so it's not clear if we should be supporting it.

**Comment by bestlem** [comment]
> > The documentation needs to also say when editing flake.nix
> > Edit the two "simple" in the end of the file to the first part of the machine name (not including any .local or other domain) m
> 
> The flake installation instructions say the following:
> 
> > Make sure to replace all occurrences of `simple` with your short hostname which you can find by running `hostname -s`.
> 
> Could you suggest how we could make this more clear for beginners?

Make the highlighting the same for the 'simple' and the change in architecture.

I don"t have a large enough screen to have the instructions beside my editor so when I switched back my eyes were drawn to the highlighted issue.

As for the nix-installer I seem tom have the broken version i reported on the bug. I just seem to have bad timing.

**Comment by bestlem** [comment]
Another way might be to number each substep - e.g. i0 for the flake creation ii0 for edit name the add not for architecture

**Comment by bestlem** [comment]
As it is I now do have in a new VM a path of 
`> echo $PATH                                                                    ~/.config/nix-darwin
/Users/dev/.nix-profile/bin:/run/current-system/sw/bin:/nix/var/nix/profiles/default/bin:/usr/local/bin:/usr/bin:/usr/sbin:/bin:/sbin`

But after a reboot - which I think is needed to create the /run entry
```
> darwin-rebuild switch --flake ~/.config/nix-darwin                                  ~/.config/nix-darwin
zsh: command not found: darwin-rebuild
> ls /run/current-system                                                              ~/.config/nix-darwin
ls: /run/current-system: No such file or directory
> sudo cat /etc/synthetic.conf                                                        ~/.config/nix-darwin
nix
run	private/var/run


```

**Comment by emilazy** [comment]
Are you sure the activation ran correctly? For instance was there an error about your `nix.conf` hash? (A reboot should not be required.)

**Comment by bestlem** [comment]
Yes there was an errro about /.etc/nix and doing a clean install again and removing that with a rerun worked.

The Deterministic Systems Installer added the /etc/nix so that gave me the issue but it is now working

Sorry for the noise here.

**Comment by emilazy** [comment]
https://github.com/LnL7/nix-darwin/pull/731 should fix the partial activation and give a clearer error in that case. Will leave this open for the docs issue.

**Comment by frencojobs** [comment]
I also installed nix using deterministic system's installer and it wrote `/etc/nix/nix.conf`. Now, nix-darwin is asking me to either remove it or rename it as a backup. These are the content of the file, as generated by the installer.

```conf
# Generated by https://github.com/DeterminateSystems/nix-installer, version 0.10.0.
build-users-group = nixbld
bash-prompt-prefix = (nix:$name)\040
auto-optimise-store = true
extra-nix-path = nixpkgs=flake:nixpkgs
experimental-features = nix-command flakes auto-allocate-uids
```

Is it safe to just remove them? I'm aware the last line will be generated by nix-darwin because it's configured in `flake.nix` like this,

```nix
# Necessary for using flakes on this system.
nix.settings.experimental-features = "nix-command flakes";
```

But, shouldn't I also add configurations like this, so that the original behavior from installer is restored?

```nix
nix.settings.build-users-group = "nixbld"
```

**Comment by Enzime** [comment]
@frencojobs [build-users-group](https://nixos.org/manual/nix/stable/command-ref/conf-file.html#conf-build-users-group) defaults to `nixbld` when using the Nix daemon, so it's not necessary to explicitly specify, but you can add it to your config if you'd like

**Comment by dstcruz** [comment]
If you comes here from having a fresh `nix` install, might be nice to point out that the `--extra-experimental-features` is needed for doing a flake install? Would be nice for simpler copy-pasta of the command, like:

```shell
> cd my-flake-directory
> nix --extra-experimental-features 'nix-command flakes' run nix-darwin -- switch --flake .#simple
```

Unless you guys think it is pretty safe to assume that if someone is trying to do a flakes based system they would already know this.

**Comment by britter** [comment]
I've also installed Nix using the Determinate Systems installer. After initializing a nix-darwin flake using `nix init flake -t nix-darwin` I ran into the problem reported by @frencojobs. I fixed by renaming `nix.conf` and running the command suggested by @dstcruz. I can now switch my system using `nix run nix-darwin -- switch --flake .`. However I don't have `darwin-rebuild` in my PATH which is why I get

```
darwin-rebuild switch --flake .
fish: Unknown command: darwin-rebuild
```

Any advice on how to get `darwin-rebuild` into PATH?

**Comment by jeeftor** [comment]
Where is nix-darwin's flake file?

**Comment by Dedoque** [comment]
So many headpain in simple action -- just install packet manager
Not easily use lamp homebrew? =)

Ive read and try minim 3 manuals to install nix-darwin.  And I wasted a lot of time pursuing his savings of time! 
And no one on the fly! Any of them on some steps returns errors. You fix one error, in next time will be return another! And any try like that!

Have 100% method for install nix-darwin on-the-fly easily? Or need to be red-eyed-core-gentoo man read and make patchs?
Don't you think it's easier to use Ansible?
